src/azure-cli-core/azure/cli/core/_util.py

```diff
@@ -108,3 +108,7 @@ def to_camelcase(s):
                      if not callable(v) and not k.startswith('_')])
     else:
         return obj
+
+def to_snake_case(s):
```

Comment:
i suggest put in couple of tests on this, particular if the name is already separated by '_', are we adding a redundant one? Of course, if that will never happen, then you can just add a simple comment.